### ISSUE TITLE
Summary of events after each turn (GNA #4167)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1042]
**[Original submission](https://gna.org/bugs/index.php?4167)** by anonymous on 2005-06-12

Would like an option to display a summary of events after each turn, and as an option, offer to replay each event.

(Reproduced on Windows XP)
Release: 1.0
Priority: 3 - Low
Severity: 3 - Normal

**Comment by wesnoth-bugs** [comment]
**Modified on 2005-10-10**

**ott wrote:**

> See https://savannah.nongnu.org/bugs/index.php?func=detailitem&amp;item_id=13384 for Savannah history.

**ott changed priority:** 1 - Later -> 3 - Low

**Comment by wesnoth-bugs** [comment]
**Modified on 2006-01-26**

**ott wrote:**

> See also bug #4966.

**Comment by wesnoth-bugs** [comment]
**Modified on 2006-01-26**

**ott wrote:**

> See also bug #3836 for a more detailed proposal for part of this.  That one may be much easier to implement -- this one looks like a huge amount of work.

**Comment by wesnoth-bugs** [comment]
**Modified on 2007-08-03**

**boucman wrote:**

> a clean way to do it would be to &quot;block&quot; the UI until the start of your next turn, then run the whole turn as a replay and, last start the clock and start playing.

**Comment by wesnoth-bugs** [comment]
**Modified on 2007-08-17**

**anonymous wrote:**

> Boucman: Sounds like fun (and useful). In addition to what you propose, if you can make the turn bell and turn dialog still work (while rest of the UI and game animations are frozen) when the game arrives back at your turn, this would be excellent.
> 
> 
> 
> To prevent annoying delay for other players in MP:
> 
> - _If timer is used,_ maybe allow for a 30 second grace period before the user has to click away the &quot;it is your turn&quot; -dialog, and if not done by that time, automatically remove it and start playing the turn replay.
> 
> - If no timer is used, the dialog can stay until clicked away.
> 
> 
> 
> - If the dialog isn't used, but only turn bell, maybe force a grace period of (say) 5 seconds, so the user has time to switch their focus to the Wesnoth window, before starting the animation?
> 
> 
> 
> - If neither the dialog or the turn bell are used, the animation should probably start playing right after the previous player ends their turn.
> 
> 
> 
> This would give the best of both worlds, that is flexibility and control over the UI for the player, and a standard no-nonsense
> 
> schedule for other MP players in a timed game.

**Comment by Pentarctagon** [comment]
Closing due to age, this already being possible by just saving a replay and going back to that turn, and since the undoubted complexity involved in getting this to work with the current number of UMC API features during a running MP game isn't worth the benefit.

**Comment by soliton-** [comment]
The replay part doesn't seem that complex. Save a snapshot at the start of the turn and save the replay commands separately as they come in. Still sounds like a nice to have feature.
